test/specs/commonTests.js

```diff
@@ -111,6 +113,67 @@ export const isConformant = (Component, requiredProps = {}) => {
   })
 
   // ----------------------------------------
+  // Events
+  // ----------------------------------------
+  it('handles events transparently', () => {
+    // Stardust events should be handled transparently, working just as they would in vanilla React.
+    // Example, both of these handler()s should be called with the same event:
+    //
+    //   <Button onClick={handler} />
+    //   <button onClick={handler} />
+    //
+    // This test catches the case where a developer forgot to call the event prop
+    // after handling it internally. It also catch cases where the synthetic event was not passed back.
+    _.each(syntheticEvent.types, ({ eventShape, listeners }) => {
+      _.each(listeners, listenerName => {
+        // onKeyDown => keyDown
+        const eventName = _.camelCase(listenerName.replace('on', ''))
+
+        // onKeyDown => handleKeyDown
+        const handlerName = _.camelCase(listenerName.replace('on', 'handle'))
+
+        const handlerSpy = sandbox.spy()
+        const props = {
+          ...requiredProps,
+          [listenerName]: handlerSpy,
+          simulateEventOnThisComponent: true,
+        }
+
+        const wrapper = shallow(<Component {...props} />)
+
+        wrapper
+          .find('[simulateEventOnThisComponent]')
+          .simulate(eventName, eventShape)
+
+        // give event listeners opportunity to cleanup
+        if (wrapper && wrapper.unmount && wrapper.instance().componentWillUnmount) {
```

Comment:
Totally agreed, the Dropdown will come in 3 PRs actually.  The first rough draft, the second adding `multiple` prop, and the third doing cleanup and adding abstractions.